tests/bsim/bluetooth/audio/prj.conf

```diff
@@ -3,6 +3,7 @@ CONFIG_BT_TESTING=y
 CONFIG_CONSOLE_HANDLER=y
 CONFIG_USERSPACE=n
 CONFIG_NO_OPTIMIZATIONS=y
+CONFIG_ZTEST_ALLOW_NO_OPTIMIZATIONS=y
```

Comment:
This is a babblesim test and is not related to ZTEST. I think you can remove this line :)